articles/055_ros_parameter_design.md

```diff
@@ -183,6 +183,16 @@ A similar behavior can be implemented by allowing the search parameter implement
 
 The client libraries will provide the following API for interfacing with the Core Parameter API for both local and remote nodes including return codes.
 
+### Parameter Initialization
+
+Default parameters initialization may be overriden via command line arguments using the `name!=value` syntax, where:
+
+- `name` is the parameter name that is to be initialized.
+- `value` is the value, in [YAML](https://yaml.org/spec/1.2/spec.html) syntax, to initialize the parameter with.
+- `!=` is the parameter overriding assignment operator.
+
+Note the use of the `!=` operator in lieu of the `:=` operator, to help disambiguate parameter initialization statements from name remapping rules.
```

Comment:
Why not splitting the remapping rules from the parameters with a separator? e.g.:
```bash
ros2 run package executable asd:=bsd --params param1:=1 param2:=asd
ros2 run package executable asd:=bsd -- param1:=1 param2:=asd
```

I slightly prefer the second option.

Reply:
I really like how brief it is.

Reply:
I'm concerned somewhat about the use of `!` as in my experience in shells it needs to be escaped and can be difficult to do so sometimes. I always avoid them in my commit messages because this 🤣.

Reply:
I like `--params` better than `--`, because you have to remember that we may be sharing the command line with users and `--` is not very explicit.

Reply:
I know it would break everything, but if I were doing this from scratch I'd use `from->to` for remapping and `key:=value` for parameters. :/

Reply:
I guess that's similar to `~=` for remapping, as you suggested. But I agree that may be too disruptive.

Reply:
Yeah. Also, won't `from->to` have issues with redirection in most shells?

Reply:
Yeah, no good options.

Reply:
I recommend against using `--` for this. It's very commonly used (e.g. by bash builtins, grep, etc.) to signify the end of CLI opions for the command, after which only positional parameters are accepted (e.g. it's very common for commands which run other commands in a subshell, as those other commands will have their own options). Seeing it used as suggested here would be pretty confusing.